e3sm_diags/driver/utils/dataset_xr.py

```diff
@@ -413,6 +413,14 @@ def _get_climo_dataset(self, season: str) -> xr.Dataset:
         # ds = ds[[self.var, 'lat_bnds', 'lon_bnds']]
         ds = ds[[self.var] + keep_bnds]
 
+        # NOTE: There seems to be an issue with `open_mfdataset()` and
+        # using the multiprocessing scheduler defined in e3sm_diags,
+        # resulting in timeouts and resource locking.
+        # To avoid this, we load the multi-file dataset into memory before
+        # performing downstream operations.
+        # Related GH issue: https://github.com/pydata/xarray/issues/3781
+        ds.load(scheduler="sync")
```

Comment:
This is needed to resolve conflicts between `multiprocessing` and dask. Though we need it here to only load variables needed, otherwise, out of memory still occur as @tomvothecoder noted.